src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -1223,11 +1224,15 @@ public String getShutdownMessage()
 		throw new UnimplementedOperationException();
 	}
 
+	public void setWarningState(@NotNull WarningState warningState)
+	{
+		this.warningState = warningState;
+	}
```

Comment:
Two remarks on this:
1. Since this method is not part of the Bukkit API specification, I reckon the method should be documented via javadocs
2. The `@NotNull` param annotation is not yet enforced, passing a null value would create an inconsistent state and violate the return type of `getWarningState()` being guaranteed to be non-null. So we should add a validation check here.

```suggestion
	public void setWarningState(@NotNull WarningState warningState)
	{
	    Validate.notNull(warningState, "The warning state cannot be null!");
		this.warningState = warningState;
	}
```